### PR TITLE
update Documentation, readme and docstrings

### PR DESCRIPTION
**Pull request by dmetivie**
I made a few update to documentations like
- fixing docstring call of `sample`. 
- Changing a bit some docstrings to match the PR #65 rebase
- Separate in a new page generating matrix for clarity
- Add an example in home page of doc of QMC vs MC (this seems like a must-have for a QMC pkg).
- Change Project.toml of docs to use [Documenters.jl code block example ](https://documenter.juliadocs.org/stable/man/syntax/#@example-block).

I hope this help!

BTW, I believe we should remove `RandomSample` sometime called `UniformSample`. Instead, we can directly use `Uniform` in `sample`.

### CODE REVIEW

**Comment by dmetivie** [comment]
I still cannot figure why format check fail, I tried JuliaFormatter every possible ways!

**Comment by ChrisRackauckas** [comment]
The formatter just had an update that changed some of the formatting. I'll update the format in a separate commit.

File: docs/src/design_matrix.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+It is often convenient to generate multiple independent sequence, for error estimation (uncertainty quantification).
+The resulting sequences can be stored in what is often called a design matrix.
+In this package, this is achieved with the `generate_design_matrices(n, d, ::DeterministicSamplingAlgorithm), ::RandomizationMethod, num_mats)` function. `num_mats` is the number of independent realization. The resulting design matrix is a vector of matrix of length `num_mats`.
+
+```@docs
+QuasiMonteCarlo.generate_design_matrices
+```
+
+!!! warning
+    The method `generate_design_matrices(n, d, sampler, R::NoRand, num_mats, T = Float64)` is an ad hoc way to produce a Design Matrix. Indeed, it creates a deterministic point set in dimension `d × num_mats` and splits it into `num_mats` point set of dimension `d`. The resulting sequences have no QMC guarantees.
```
Comment (ChrisRackauckas) [inline docs/src/design_matrix.md:15]: Interesting. Is there a better way to do this which guarantees the good properties for QMC?

File: docs/src/design_matrix.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    This seems to have been proposed in Section 5.1 of [*Saltelli, A. et al. (2010)*](https://d1wqtxts1xzle7.cloudfront.net/76482087/PUBLISHED_PAPER-libre.pdf?1639660270=&response-content-disposition=inline%3B+filename%3DVariance_based_sensitivity_analysis_of_m.pdf&Expires=1685981169&Signature=aim5tHldlkb0ewZ9-gSMZsW2F1b88tLvV8euV1FpD61UYrE1mLR3RDERut0BsHNbcibjKQnF1JlsZ8mtEx~E1~eI3A~SOSySbpQllIpbhu556pFGUvD3GV5M6ghwa-5QMDP3-aQczBzflR721N4PCVJgqfmV-y94pkijQYvHSvZaPKb-tsoS8TVxE6H31Ptk4u662H61ofKzXR5JCHv0740qkQ0hORH~GqXOt8s7yQMVWYswZT4pWGSkJ9EehEQHCLo2uDVW-YSopwlSaaMRz~~0O~hkGAVE8sC~TAB7b5KnUgtNXl0jYTfGNTYO4GNJo1XhmHwj~Og~sBLDIXDxsg__&Key-Pair-Id=APKAJLOHF5GGSLRBV4ZA)[^1] to do uncertainty quantification.
```
Comment (ChrisRackauckas) [inline docs/src/design_matrix.md:16]: Yes that is the source.

File: docs/src/randomization.md [hunk 3]
```diff
@@ -1,96 +1,140 @@
...
-`Shift(rng)` a.k.a. Cranley Patterson Rotation. It is by far the fastest method, it is used in `LatticeRuleScramble` for example.
+- `DigitalShift` the simplest and faster method. For a point $x\in [0,1]^d$ it does $y_k = (x_k + U_k) \mod b$ where $U_k \sim \mathbb{U}(\{0, \cdots, b-1\})$
+
 ```@docs
-  Shift
+DigitalShift
 ```
 
-## Design Matrices
+- `MatousekScramble` a.k.a. Linear Matrix Scramble is what people use in practice. Indeed, the observed performances are similar to `OwenScramble` for a lesser numerical cost.
```
Comment (ChrisRackauckas) [inline docs/src/randomization.md:76]: Is there a quick description of this one?

**Comment by ChrisRackauckas** [comment]
This is amazingly clear. Thanks for putting the time in. I left some small comments, but they are small so I'm going to merge and try the new formatter.

**Pull request merged**